src/main/java/wayoftime/bloodmagic/util/helper/NetworkHelper.java

```diff
@@ -33,16 +36,14 @@ public class NetworkHelper
 	 */
 	public static SoulNetwork getSoulNetwork(String uuid)
 	{
-		if (dataHandler == null)
-		{
-			if (ServerLifecycleHooks.getCurrentServer() == null)
-				return null;
+		if (ServerLifecycleHooks.getCurrentServer() == null)
+			return null;
 
-			DimensionDataStorage savedData = ServerLifecycleHooks.getCurrentServer().overworld().getDataStorage();
+		DimensionDataStorage savedData = ServerLifecycleHooks.getCurrentServer().overworld().getDataStorage();
 
+		if (dataHandler == null || cachedDataStorage != savedData)
```

Comment:
`cachedDataStorage` is never assigned a value, so this is effectively `if (dataHandler == null || savedData != null)`.